core/io/config_file.cpp

```diff
@@ -61,80 +61,155 @@ PackedStringArray ConfigFile::_get_section_keys(const String &p_section) const {
 	return arr;
 }
 
+void ConfigFile::_create_entry_if_needed(const String &p_section, const String &p_key) {
+	if (!entries.has(p_section)) {
+		entries.insert(p_section, HashMap<String, Entry>(), p_section.is_empty());
+	}
+	if (!entries[p_section].has(p_key)) {
+		entries[p_section].insert(p_key, Entry());
+	}
+}
+
+String ConfigFile::_get_entry_as_string(const String &p_key, const Entry &p_entry) const {
+	String estr;
+
+	if (!p_entry.comment.is_empty()) {
+		PackedStringArray lines = p_entry.comment.split("\n", true); // allow empty for multi-line separation in detailed comments
+		
+		// marker for removing the empty front lines
+		int first_populated_line = -1;
+
+		// remove empty lines in the back
+		for (int i = lines.size() - 1; i > -1; i--) {
+			if (lines[i].strip_edges().is_empty()) {
+				if (first_populated_line == -1) {
+					lines.remove_at(i);
+				}
+			} else {
+				first_populated_line = i;
+			}
+		}
+		// remove empty lines in the front
+		for (int i = first_populated_line; i > -1; i--) {
+			if (lines[i].strip_edges().is_empty()) {
+				lines.remove_at(i);
+			}
+		}
+
+		if (!lines.is_empty()) {
+			for (int i = 0; i < lines.size(); i++) {
+				String line = lines[i].strip_edges(false, true); // no reason to keep the right edge i think?
+				estr += ("; " + line + "\n"); // it should be ok to always indent the comment, right? maybe make the auto-indentation a project setting...
```

Comment:
These two comments do not feel necessary
```suggestion
				String line = lines[i].strip_edges(false, true);
				estr += ("; " + line + "\n");
```